lib/runner.js

```diff
@@ -32,14 +32,42 @@ function parseExitCode (buffer, defaultExitCode, failOnEmptyTestSuite) {
 }
 
 // TODO(vojta): read config file (port, host, urlRoot)
-function run (config, done) {
-  config = config || {}
-
-  logger.setupFromConfig(config)
-
+function run (cliOptionsOrConfig, done) {
+  cliOptionsOrConfig = cliOptionsOrConfig || {}
   done = helper.isFunction(done) ? done : process.exit
-  config = cfg.parseConfig(config.configFile, config)
 
+  let config
+  if (cliOptionsOrConfig instanceof cfg.Config) {
+    config = cliOptionsOrConfig
+  } else {
+    // TODO: Should `const log = logger.create('runner')` be moved into the
```

Comment:
nit: I'm fine with either, including letting it be inconsistent. So choose whichever option and remove this TODO.

Otherwise, LGTM to start working on tests.

Reply:
I'll remove the comment and begin work on the tests.

<details>

The code in question is the use of this function

https://github.com/karma-runner/karma/blob/d3ff91a5a2f7ff84141096bb47aa31ff73f81cca/lib/logger.js#L88-L106

My preference is to let the code speak for/document itself. In this case, there is nothing instance specific about the result of `logger.create`. My call would normally be to use it once at the module level, so `server.js` and `stopper.js` would be updated.

I don't expect the change would break anything, but in the spirit of expecting the unexpected, I'm going to leave it alone for now, delete the comment, and let it remain inconsistent. This PR should remained focused on promise support. Changing the logger statement, however minor it appears to be, falls under the category of general maintenance or logger improvements. As long as someone is aware of it, then the comment being removed accomplished its goal.

Since it pulls from a cache AND none of karma's internal code appears to use the second argument, there is effectively no difference between using it at the constructor/function level vs once at the module level.

</details>

Reply:
I agree that it should not affect anything. Feel free to send a separate PR to make it consistent. Maybe there are other places where we have a similar inconsistency. Having the call at the module level makes sense to me.